lectures/zksnark.tex

```diff
@@ -0,0 +1,271 @@
+\documentclass[../lecture-notes.tex]{subfiles}
+
+\begin{document}
+
+\subsection{What the zk-SNARK is?}
+
+Let's first discuss what zk-SNARK is.
+
+\begin{definition}
+    \textbf{zk-SNARK} – Zero-Knowledge Succinct Non-interactive ARgument of Knowledge.
+\end{definition}
+
+But what do terms like "argument of knowledge," "succinct," "non-interactive," and 
+"zero-knowledge" mean in this context?
+
+\begin{itemize}
+    \item \textbf{Argument of Knowledge} - a proof that the prover know data that resolves a certain
+    problem, and this knowledge can be verified.
+    \item \textbf{Succinct} - the proof size is relatively small and does not depend on the size of 
```

Comment:
Specify that you will show what "relatively small" means specifically in the further discussion